components/AvalancheForecastZoneMap.tsx

```diff
@@ -197,13 +242,105 @@ class AnimatedDrawerController {
+    const projectionConversionFactor = (latitudeDegrees: number): number => {
+      return 1 / Math.cos((latitudeDegrees * Math.PI) / 180);
+    };
+
+    // next, we need to figure out if our constraining behavior is fitting the polygons in width-wise, or height-wise
+    // by comparing the aspect ratios of our bounding region and the unobstructed view.
+    const regionAspectRatio = this.baseRegion.longitudeDelta / (projectionConversionFactor(this.baseRegion.latitude) * this.baseRegion.latitudeDelta);
+    const viewAspectRatio = unobstructedWidth / unobstructedHeight;
+
+    // next, we determine the conversion factor between pixels and geographic coordinate degrees
+    let degreesPerPixelHorizontally = 0;
+    let degreesPerPixelVertically = 0;
+    if (regionAspectRatio > viewAspectRatio) {
+      // our region is wider than our view, so width is our limiting factor
+      degreesPerPixelHorizontally = this.baseRegion.longitudeDelta / unobstructedWidth;
+      degreesPerPixelVertically = degreesPerPixelHorizontally / projectionConversionFactor(this.baseRegion.latitude);
+    } else {
+      // our region is taller than our view, so height is our limiting factor
+      degreesPerPixelVertically = this.baseRegion.latitudeDelta / unobstructedHeight;
+      degreesPerPixelHorizontally = degreesPerPixelVertically * projectionConversionFactor(this.baseRegion.latitude);
+    }
+
+    // knowing these conversion factors, we can calculate the size of the bounded region in the view
+    const regionWidthPixels = this.baseRegion.longitudeDelta / degreesPerPixelHorizontally;
+    const regionHeightPixels = this.baseRegion.latitudeDelta / degreesPerPixelVertically;
+
+    // then, we can determine the location of the center of the bounded region in the view
+    // the center in X will be:
+    const regionCenterXPixel =
+      regionWidthPixels / 2 + // half the size of the region
+      (this.windowWidth - regionWidthPixels) / 2; // half of the leftover space in the screen
+    // the center in Y will be:
+    const regionCenterYPixel =
+      this.topElementsHeight + // the top elements
+      regionHeightPixels / 2 + // half the size of the region
+      (this.windowHeight - this.topElementsHeight - cardDrawerHeight - this.tabBarHeight - regionHeightPixels) / 2; // half the leftover space in the screen
+    // finally, we calculate the offset, in pixels, from the center of the region to the center of the screen
+    const regionCenterXOffsetPixels = regionCenterXPixel - this.windowWidth / 2;
+    const regionCenterYOffsetPixels = regionCenterYPixel - this.windowHeight / 2;
+
+    // now, we can position our region relative to our overlays by asking the displayed region to:
+    // - center around the virtual point that results in the correct layout
+    // - contain enough longitude and latitude deltas to match the whole screen
+    const displayedRegion: Region =
+      degreesPerPixelVertically > 0
```

Comment:
We need this funky check as this method fires in the first couple of render passes when we still don't know the window dimensions, the size of the tab bar, etc. Open to better approaches here!